src/fetchExceptions.ts

```diff
@@ -1,20 +1,39 @@
-import {
-  Response as FetchResponse
-} from 'node-fetch'
+import { Response as FetchResponse } from 'node-fetch'
 
+export const checkStatusCode = (
+  response: FetchResponse
+): number | undefined => {
+  let { hostname } = new URL(response.url)
+
+  let code: number | undefined
+  let nextHostname: string | undefined
+  while (code == null && hostname !== nextHostname) {
+    hostname = hostname ?? nextHostname
```

Comment:
It appears this will infinitely loop. Did you mean to do

`hostname = nextHostname ?? hostname`